docs/design-docs/01-kpt-live-stdin.md

```diff
@@ -0,0 +1,242 @@
+# Title
+
+* Author(s): Yuwen Ma <yuwenma@google.com>
+* Contributor(s):
+    - Sean Sullivan <seans@google.com>
+    - Sunil Arora <sunilarora@google.com>
+    - Mortent Torkildsen <mortent@google.com>
+    - Mengqi Yu <mengqiy@google.com>
+    - Natasha Sarkar <natashasarkar@google.com>
+
+* Approver: Sunil Arora <sunilarora@google.com>
+
+## Why
+
+This design is aimed at making `kpt live` be hydration-tool-neutral so that `kpt live`
+applicable use cases can be expanded to `kustomize build`, `helm templates` and raw
+kubernetes manifests.
+
+### Non-goal
+- This doc is not aimed at designing a unified hydration between kpt and kustomize
+- This doc is not aimed at changing the kpt in-place hydration models.
+
+## Design
+
+We propose to make `kpt live` support reading configurations from standard input,
+detach the Inventory from Kptfile and simplify the Inventory-ResourceGroup mechanism
+to use  ResourceGroup only.
+
+### Read standard input resources
+
+We will add a reserved keyword “-” as a special kpt directory. This "-" accepts resources
```

Comment:
Does that not already exist?

Reply:
Yes, `kpt live apply -` [already exists](https://github.com/GoogleContainerTools/kpt/blob/3b42daa41102fa83bce07bd82a72edcd691868d6/internal/cmdapply/cmdapply.go#L52) and [works with STDIN](https://github.com/GoogleContainerTools/kpt/blob/3b42daa41102fa83bce07bd82a72edcd691868d6/pkg/live/load.go#L72). It does require a Kptfile with Inventory information in the stream.

Reply:
updated.